### ISSUE TITLE
Fan Adjustment

### ISSUE THREAD

**Issue by MlgmXyysd** [BartoszCichecki/LenovoLegionToolkit#1028]
### Rules

- [X] I made myself familiar with the README.
- [X] I checked closed issues for similar ones and did not find a similar feature request mentioned before.
- [X] I understand that, if insufficient information was provided, my issue will be closed without an answer.

### Version

2.16.2

### OS

Windows 11 22H2 build 22621.2506

### Device

Legion Y9000P 2023 IAH7H

### Is your feature request related to a problem?

In Lenovo PC Manager, there is a feature that allows the cooling fan to run at full speed for a period of time, which is very useful for temporary cooling. I know I can adjust the fan run strategy in custom mode. However, custom mode does not work without being connected to the stock power supply. Need a similar feature that can quickly reduce computer temperatures.

### How would you like the problem to be solved?

None

### What alternatives have you considered?

None

### Additional information

None

**Comment by BartoszCichecki** [comment]
I see, what is the point of using this function? Legions cools down very quickly, so why bother with max fans? In any case if you need to use max fans, it sounds like your fans needs some cleaning.

**Comment by MlgmXyysd** [comment]
> I see, what is the point of using this function? Legions cools down very quickly, so why bother with max fans? In any case if you need to use max fans, it sounds like your fans needs some cleaning.

Usually it dissipates heat quickly, but sometimes not. Especially when running some software and games.
These days I've been experiencing a lot of overheating of my computer, causing performance drops or crashes. It has improved after running computer manager's fan at full speed several times.

**Comment by zzhzero** [comment]
> I see, what is the point of using this function? Legions cools down very quickly, so why bother with max fans? In any case if you need to use max fans, it sounds like your fans needs some cleaning.

like this https://www.legionfancontrol.com/

Customizing the relationship between speed and temperature is indeed a necessary requirement

**Comment by d3adwolf** [comment]
> I see, what is the point of using this function? Legions cools down very quickly, so why bother with max fans? In any case if you need to use max fans, it sounds like your fans needs some cleaning.

Hi, I have a Lenovo Legion 7i Slim with RTX 4070 and I really need a button in Toolkit for max RPM because my left CPU fan is only spinning 4100/4700 RPM while the GPU is spinning 4600/4700, why this happens is not clear. I can through Custom Profile make it run at 4700 RPM, but there are wrong settings and the CPU drops frequencies. I'm waiting for a new Honeywell thermal paste, but at the moment I have a good thermal paste and a clean fan, but the CPU temperatures (which is Core i9-13900H) are close to 100 Celsius, 97 to be exact and I'm surprised the RPMs are 4100. So it would be nice to have on Perfomance Profile (red indicator) just turning the RPMs up to 4700 RPM. Thanks in advance for both the reply and all the work on the application.

**Comment by d3adwolf** [comment]
I will wait for a reply to my problem on the [forum](https://forums.lenovo.com/t5/Gaming-Laptops/Legion-Slim-7-16IRH8-fan-speed-is-not-maximal/m-p/5268574).

**Comment by BartoszCichecki** [comment]
> > I see, what is the point of using this function? Legions cools down very quickly, so why bother with max fans? In any case if you need to use max fans, it sounds like your fans needs some cleaning.
> 
> 
> 
> like this https://www.legionfancontrol.com/
> 
> 
> 
> Customizing the relationship between speed and temperature is indeed a necessary requirement
> 
> 
> 
> 

LegionFanControl controls fans by directly manipulating EC memory. This is neither safe nor reliable and will not be part of LLT.

**Comment by BartoszCichecki** [comment]
> These days I've been experiencing a lot of overheating of my computer, causing performance drops or crashes. It has improved after running computer manager's fan at full speed several times.

That sounds like a bigger issue and using max fans is just a workaround that will put unnecessary load on the fans.

**Comment by Frosty-J** [comment]
> LegionFanControl controls fans by directly manipulating EC memory. This is neither safe nor reliable and will not be part of LLT.

Yeah, just had a look at it myself. Windows warns it's malicious when running the installer. Then memory integrity and the vulnerable driver blocklist need disabling to use it. Seems like it doesn't work on the LOQ - no fan curve table found. Would be nice to have, but if the same steps are required it seems out of scope for LenovoLegionToolkit which just needs admin.

**Comment by dustojnikhummer** [comment]
> > LegionFanControl controls fans by directly manipulating EC memory. This is neither safe nor reliable and will not be part of LLT.
> 
> Yeah, just had a look at it myself. Windows warns it's malicious when running the installer. Then memory integrity and the vulnerable driver blocklist need disabling to use it. Seems like it doesn't work on the LOQ - no fan curve table found. Would be nice to have, but if the same steps are required it seems out of scope for LenovoLegionToolkit which just needs admin.

I know the creator of LFC. LFC is now considered EOL. As you said, it only supports 2021 and 2022 models and uses rwdrv.sys, which we know is flagged as Defender. Afaik LLT can control fans on some models, but that is only on laptops supported by Lenovo Legion Arena. Give that a shot and see if your LOQ is supported.

**Comment by kolosovpetro** [comment]
+1 to this feature request tho

**Comment by rango886** [comment]
https://github.com/johnfanv2/LenovoLegionLinux This project can control the fan of the Legion 7 16ACHg6 2021/r9000k, this project is controls fans by directly manipulating EC memory too?

**Comment by rango886** [comment]
I found a project https://github.com/SmokelessCPUv2/Lagon-Fan-EC-Control
It write by c#, directly manipulating EC memory, It has a similar interface to LenovoLegionToolkit
This project supports legion 2022/2021 model, It is work on my Legion 7 16ACHg6 2021/r9000k

**Comment by rango886** [comment]
This project https://github.com/SmokelessCPUv2/Lagon-Fan-EC-Control  is great !!!!
This project can write the curve directly to the EC, turn off the software or restart laptop, and the fan control will still work
But when change mode(fn + q), fan curve wil be reset to default

**Comment by rango886** [comment]
I've written a command-line tool that supports controlling legion fan speed on Windows !!! https://github.com/rango886/LegionFanEmbeddedController

Use this library https://github.com/GermanAizek/WinRing0  to access EC memory, written in C#, run with only administrator privileges
I'm not very good at gui programming, and hopefully someone can integrate my project into the LenovoLegionToolkit

**Comment by BartoszCichecki** [comment]
@rango886 congrats on making the app. However as I said before using ring0 drivers isn't safe and therefore it will not be included in LLT.

**Comment by svensvond** [comment]
I would even pay money if i was able to control the fans on 2023/2024 models like i can with LegionFanControl on 2021 models.

**Comment by k3nsh1nta** [comment]
![Screenshot 2024-12-16 002848](https://github.com/user-attachments/assets/b655c3ed-ad23-4f90-b588-821550778d05)

Is this a bug or what?

**Comment by svensvond** [comment]
does it work?
